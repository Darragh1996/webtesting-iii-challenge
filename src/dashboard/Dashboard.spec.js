// Test away
import React from "react";
import ReactDom from "react-dom";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Dashboard from "./Dashboard";

afterEach(rtl.cleanup);

let Unlocked = () =>{
    return wrapper.queryByText('Unlocked');
}
let Open = () =>{
    return wrapper.queryByText('Open');
}
let LockGate = () =>{
    return wrapper.queryByText('Lock Gate');
}
let CloseGate = () =>{
    return wrapper.queryByText('Close Gate');
}
let Closed = () =>{
    return wrapper.queryByText('Closed');
}
let Locked = () =>{
    return wrapper.queryByText('Locked');
}
let OpenGate = () =>{
    return wrapper.queryByText('Open Gate');
}
let UnlockGate = () =>{
    return wrapper.queryByText('Unlock Gate');
}

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(<Dashboard />);
});

it("renders without crashing", () => {
  expect(wrapper.container).toMatchSnapshot();
});
