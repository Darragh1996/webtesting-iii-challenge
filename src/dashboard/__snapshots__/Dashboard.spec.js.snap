// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expected elements properly -> AFTER CLOSE GATE BUTTON IS CLICKED 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Open Gate
    </button>
  </div>
</div>
`;

exports[`renders expected elements properly -> AFTER CLOSE GATE WAS CLICKED AND LOCK GATE WAS THEN CLICKED 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led red-led"
    >
      Locked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
    >
      Unlock Gate
    </button>
    <button
      class="toggle-btn"
      disabled=""
    >
      Open Gate
    </button>
  </div>
</div>
`;

exports[`renders expected elements properly-> DEFAULT 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;
